controllers/custodian/add.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 SAP SE or an SAP affiliate company
```

Comment:
We can again rename the file to manager.go

Reply:
ok, sure

Reply:
In this file we will only have functions which will add a reconciler to the manager. We have borrowed this convention from g/g. Out of several examples of such usage there, you can see one usage at https://github.com/gardener/gardener/tree/master/pkg/controller/service

Reply:
register.go seems fine

Reply:
"add" and "register" are synonymously used terms for the same operation - adding a controller to a controller manager. We borrowed the term "add" from g/g (ex: https://github.com/gardener/gardener/tree/master/pkg/controllermanager/controller/bastion) which does the same thing as we're doing here.

If we decide to rename to "register.go", it would then be required to rename the method `Reconciler.AddToManager()` to `Reconciler.RegisterWithManager()` for all the controllers as well, for semantic correctness. @unmarshall WDYT?

Reply:
That is the reason i have not yet changed it. I have personally no problem with the name `add.go` as it goes well with `AddToManager`

Reply:
@shreyas-s-rao - 2 reviewers prefer `RegisterWithManager` and `register.go`. I do not have any strong opinion on this and i am ok with either of the names. Should we then make this change?

Reply:
Ok, in that case let's make the changes.